lib/cocoapods-core/dependency.rb

```diff
@@ -109,12 +104,8 @@ def initialize(name = nil, *requirements)
         end
 
       elsif requirements.last == :head
-        @head = true
-        requirements.pop
-        unless requirements.empty?
-          raise Informative, 'A `:head` dependency may not specify version ' \
-            "requirements (#{name})."
-        end
+        raise Informative, '`:head` dependencies have been removed. Please use ' \
+          "normal external source dependencies instead of `:head` for `#{name}`."
```

Comment:
Should we improve this message a little bit more? I couldn't clearly understand what to use instead when I read the message here. Maybe we should include an example like  `:git => 'REPO_PATH'` inside the message?